api.go

```diff
@@ -362,3 +403,59 @@ func (awssf *AWSServiceFactory) GetLightsailService(regionName string) *Lightsai
 		Client: client,
 	}
 }
+
+// GetEKSService returns an instance of an EKSService associated
+// with our session. The caller can supply an optional region name to contruct
+// an instance associated with that region.
+func (awssf *AWSServiceFactory) GetEKSService(regionName string) *EKSService {
+	// Construct our service client
+	var client eksiface.EKSAPI
+	if regionName == "" {
+		client = eks.New(awssf.Session)
+	} else {
+		client = eks.New(awssf.Session, aws.NewConfig().WithRegion(regionName))
+	}
+
+	return &EKSService{
+		Client: client,
+	}
+}
+
+// Create a K8 client
+// Reused code: https://stackoverflow.com/questions/60547409/unable-to-obtain-kubeconfig-of-an-aws-eks-cluster-in-go-code
+func (awssf *AWSServiceFactory) GetK8Service(cluster *eks.Cluster) *K8Service {
+	gen, err := token.NewGenerator(true, false)
+	if err != nil {
+		return nil
+	}
+
+	tok, err := gen.GetWithOptions(&token.GetTokenOptions{
+		ClusterID: aws.StringValue(cluster.Name),
+		Session:   awssf.Session,
+	})
+	if err != nil {
+		return nil
+	}
+
+	ca, err := base64.StdEncoding.DecodeString(aws.StringValue(cluster.CertificateAuthority.Data))
+	if err != nil {
+		return nil
+	}
```

Comment:
**Thought _(non-blocking)_:**

This block would be a good candidate for a helper func.

Reply:
For this suggestion, is this what you mean?
```
func (awssf *AWSServiceFactory) GetK8Service(cluster *eks.Cluster) *K8Service {
	tok, ca, err := getTokenAndCACert(cluster, awssf.Session)
	if err != nil {
		return nil
	}

	clientset, err := kubernetes.NewForConfig(
		&rest.Config{
			Host:        aws.StringValue(cluster.Endpoint),
			BearerToken: tok.Token,
			TLSClientConfig: rest.TLSClientConfig{
				CAData: ca,
			},
		},
	)
	if err != nil {
		return nil
	}

	return &K8Service{
		Client: clientset,
	}
}

func getTokenAndCACert(cluster *eks.Cluster, session *session.Session) (token.Token, []byte, error) {
	gen, err := token.NewGenerator(true, false)
	if err != nil {
		return token.Token{}, nil, err
	}

	tok, err := gen.GetWithOptions(&token.GetTokenOptions{
		ClusterID: aws.StringValue(cluster.Name),
		Session:   session,
	})
	if err != nil {
		return token.Token{}, nil, err
	}

	ca, err := base64.StdEncoding.DecodeString(aws.StringValue(cluster.CertificateAuthority.Data))
	if err != nil {
		return token.Token{}, nil, err
	}

	return tok, ca, nil
}
```

Reply:
Yep, that works great! Another option is to have this func on the AWSServiceFactory struct too, so you don't have to pass the session -- but both work great!